client/src/components/Board/Board.tsx

```diff
@@ -0,0 +1,97 @@
+import { useContext } from 'react';
+import { BoardContext } from '../../context/useBoardContext';
+import Column from '../Column/Column';
+import { DragDropContext, Droppable } from 'react-beautiful-dnd';
+import useStyles from './useStyles';
+
+export default function Board(): JSX.Element {
+  const styles = useStyles();
+  const [board, setBoard] = useContext(BoardContext);
+
+  const onDragEnd = (result: any) => {
+    const { destination, source, draggableId, type } = result;
+    if (!destination) {
+      return;
+    }
+    if (destination.droppableId === source.droppableId && destination.index === source.index) {
+      return;
+    }
+
+    if (type === 'column') {
+      const newColumnOrder = Array.from(board.columnOrder);
+      newColumnOrder.splice(source.index, 1);
+      newColumnOrder.splice(destination.index, 0, draggableId);
+
+      const newState = {
+        ...board,
+        columnOrder: newColumnOrder,
+      };
+      setBoard(newState);
+      return;
+    }
+
+    const start = board.columns[source.droppableId];
+    const finish = board.columns[destination.droppableId];
+
+    if (start === finish) {
+      const column = board.columns[source.droppableId];
+      const newCardIds = Array.from(column.cardIds);
+      newCardIds.splice(source.index, 1);
+      newCardIds.splice(destination.index, 0, draggableId);
+
+      const newColumn = {
+        ...column,
+        cardIds: newCardIds,
+      };
+      const newState = {
+        ...board,
+        columns: {
+          ...board.columns,
+          [newColumn.id]: newColumn,
+        },
+      };
+      setBoard(newState);
+      return;
+    }
+    // list to list
+    const startCardIds = Array.from(start.cardIds);
+    startCardIds.splice(source.index, 1);
+    const newStart = {
+      ...start,
+      cardIds: startCardIds,
+    };
+    const finishCardIds = Array.from(finish.cardIds);
+    finishCardIds.splice(destination.index, 0, draggableId);
+    const newFinish = {
+      ...finish,
+      cardIds: finishCardIds,
+    };
+    const newState = {
+      ...board,
+      columns: {
+        ...board.columns,
+        [newStart.id]: newStart,
+        [newFinish.id]: newFinish,
+      },
+    };
+    setBoard(newState);
+    return;
+  };
+
+  return (
+    <DragDropContext onDragEnd={onDragEnd}>
+      <Droppable droppableId="all-columns" direction="horizontal" type="column">
+        {(provided) => (
+          <div className={styles.container} {...provided.droppableProps} ref={provided.innerRef}>
```

Comment:
Please use MUI `Box` component instead of using a `div`

Reply:
so I tried changing the div into Box component but ts is giving error
Property 'ref' does not exist on type 'IntrinsicAttributes & BoxProps & { children?: ReactNode; }'
I can update add the ref prop to Box but that would mean update it everytime.
They fixed this issue on v5 but we are using v4. any suggestions?

Reply:
This is related to the React Beautiful Drag and Drop, we can just use div here in order to comply with the library. That's more important 😄